camel/agents/chat_agent.py

```diff
@@ -181,6 +181,16 @@ def __init__(
             tool.get_function_name(): tool.func for tool in all_tools
         }
 
+        # If the user hasn't configured tools in `BaseModelBackend`,
+        # the tools set from `ChatAgent` will be used.
```

Comment:
Should we still allow users to configure the tools through the model? Also, should we allow user to also configure the `tool_choice` through the agent?

Reply:
Hey @WHALEEYE , the reason for keeping the tool in this PR within the model is to maintain support for users who only use the model. This way, the use of the model and the agent can be decoupled, since for now if usr config `tool_choice` = `required` would lead to infinity loop, we'd better to add this after this argument is well supported

Reply:
Sure, that makes sense to me